frontend/server/src/main/java/org/pytorch/serve/job/RestJob.java

```diff
@@ -136,29 +154,42 @@ private void responseInference(
          * by external clients.
          */
         if (ctx != null) {
-            MetricAggregator.handleInferenceMetric(
-                    getModelName(), getModelVersion(), getScheduled() - getBegin(), inferTime);
-            NettyUtils.sendHttpResponse(ctx, resp, true);
+            if (numStreams == 0) { // non-stream response
+                MetricAggregator.handleInferenceMetric(
+                        getModelName(), getModelVersion(), getScheduled() - getBegin(), inferTime);
+                NettyUtils.sendHttpResponse(ctx, resp, true);
+            } else if (numStreams == -1) { // the last response in a stream
+                MetricAggregator.handleInferenceMetric(
+                        getModelName(), getModelVersion(), getScheduled() - getBegin(), inferTime);
+                ctx.writeAndFlush(new DefaultHttpContent(Unpooled.wrappedBuffer(body)));
+                ctx.writeAndFlush(LastHttpContent.EMPTY_LAST_CONTENT);
```

Comment:
Minor/Curious: Instead of writing twice, can you just write once and tell netty that this is the last chunk. Also assuming you cant do that, do you need to `writeAndFlush` here or can you just write knowing that you are immediately `writeAndFlush`'ing the `LastHttpConent` chunk after?

Reply:
it needs call writeAndFlush twice, kind of similar as grpc (onComplete).